yt/visualization/volume_rendering/tests/test_off_axis_SPH.py

```diff
@@ -0,0 +1,286 @@
+    buf1 = OffAP.off_axis_projection(ds,
+                                     center,
+                                     normal_vector,
+                                     width,
+                                     resolution,
+                                     ('gas', 'density')  
+                                     )
+    find_compare_maxima(expected_maxima, buf1, resolution, width)
+
+
+@requires_module('scipy')
+def test_center_2():
+    """ Change the center to [0, -1, 0] 
+    Every point will be shifted by 1 in the y-domain
+    With this, we should not be able to see any of the y-axis particles
+    (0, 1, 0) -> (0, 2)
+    (0, 2, 0) -> (0, 3)
+    (0, 0, 1) -> (0, 1)
+    (0, 0, 2) -> (0, 1)
+    (0, 0, 3) -> (0, 1)
+    (0, 0, 0) -> (0, 1)
+    (1, 0, 0) -> (1, 1)
+    """
+    expected_maxima = ([0., 0., 0., 1.], [2., 3., 1., 1.])
+    normal_vector = [0., 0., 1.]
+    resolution = (64, 64)
+    ds = fake_sph_orientation_ds()
+    left_edge = ds.domain_left_edge
+    right_edge = ds.domain_right_edge
+    center = np.array([0., -1., 0.])
+    width = (right_edge - left_edge)      
+    buf1 = OffAP.off_axis_projection(ds,
+                                     center,
+                                     normal_vector,
+                                     width,
+                                     resolution,
+                                     ('gas', 'density')  
+                                     )
+    find_compare_maxima(expected_maxima, buf1, resolution, width)
+
+
+@requires_module('scipy')
+def test_center_3():
+    """ Change the center to the left edge, or [0, -8, 0] 
+    Every point will be shifted by 8 in the y-domain
+    With this, we should not be able to see anything !
+    """
+    expected_maxima = ([], [])
+    normal_vector = [0., 0., 1.]
+    resolution = (64, 64)
+    ds = fake_sph_orientation_ds()
+    left_edge = ds.domain_left_edge
+    right_edge = ds.domain_right_edge
+    center = np.array([0., -1., 0.])
+    width = [(right_edge[0] - left_edge[0]),
+             left_edge[1],
+             (right_edge[2] - left_edge[2])]    
+    buf1 = OffAP.off_axis_projection(ds,
+                                     center,
+                                     normal_vector,
+                                     width,
+                                     resolution,
+                                     ('gas', 'density')  
+                                     )
+    find_compare_maxima(expected_maxima, buf1, resolution, width)
+
+
+@requires_module('scipy')
+def find_compare_maxima(expected_maxima, buf, resolution, width):
+    buf_ndarray = buf.ndarray_view()
+    max_filter_buf = ndimage.filters.maximum_filter(buf, size=5)
+    maxima = np.isclose(max_filter_buf, buf_ndarray, rtol=1e-09)
+
+    # ignore contributions from zones of no smoothing
+    for i in range(len(maxima)):
+        for j in range(len(maxima[i])):
+            if np.isclose(buf[i, j], 0., 1e-09):
```

Comment:
The tests will be *a lot* faster if you use `buf_ndarray` here instead of `buf`.

Reply:
I'll give it a shot